lib/search.bash

```diff
@@ -1,50 +1,165 @@
+#!/usr/bin/env bash
 #
 # Search by Konstantin Gredeskoul «github.com/kigster»
 #———————————————————————————————————————————————————————————————————————————————
 # This function returns list of aliases, plugins and completions in bash-it,
 # whose name or description matches one of the search terms provided as arguments.
 #
 # Usage:
-#    ❯ bash-it search term1 [[-]term2] ... [[-]termN] [ --enable | --disable ]
+#    ❯ bash-it search [-|@]term1 [-|@]term2 ... \
+#       [ --enable  | -e ] \
+#       [ --disable | -d ] \
+#       [ --refresh | -r ]
+#       [ --help    | -h ]
 #
-# Exmplanation:
 #    Single dash, as in "-chruby", indicates a negative search term.
 #    Double dash indicates a command that is to be applied to the search result.
-#    At the moment only --enable and --disable are supported.
+#    At the moment only --help, --enable and --disable are supported.
+#    An '@' sign indicates an exact (not partial) match.
 #
 # Examples:
 #    ❯ bash-it search ruby rbenv rvm gem rake
-#    aliases     :  bundler
-#    plugins     :  chruby chruby-auto rbenv ruby rvm
-#    completions :  gem rake
-
+#          aliases:  bundler
+#          plugins:  chruby chruby-auto ruby rbenv rvm ruby
+#      completions:  rvm gem rake
+#
 #    ❯ bash-it search ruby rbenv rvm gem rake -chruby
-#    aliases     :  bundler
-#    plugins     :  rbenv ruby rvm
-#    completions :  gem rake
+#          aliases:  bundler
+#          plugins:  ruby rbenv rvm ruby
+#      completions:  rvm gem rake
 #
 # Examples of enabling or disabling results of the search:
 #
 #    ❯ bash-it search ruby
-#    aliases  =>   bundler
-#    plugins  =>   chruby chruby-auto ruby
+#          aliases:  bundler
+#          plugins:  chruby chruby-auto ruby
 #
 #    ❯ bash-it search ruby -chruby --enable
-#    aliases  =>   ✓bundler
-#    plugins  =>   ✓ruby
+#          aliases:  bundler
+#          plugins:  ruby
 #
+# Examples of using exact match:
+
+#    ❯ bash-it search @git @ruby
+#          aliases:  git
+#          plugins:  git ruby
+#      completions:  git
 #
+export BASH_IT_GREP=${BASH_IT_GREP:-$(which egrep)}
+declare -a BASH_IT_COMPONENTS=(aliases plugins completions)
+
 _bash-it-search() {
   _about 'searches for given terms amongst bash-it plugins, aliases and completions'
   _param '1: term1'
   _param '2: [ term2 ]...'
-  _example '$ _bash-it-search ruby rvm rake bundler'
+  _example '$ _bash-it-search @git ruby -rvm rake bundler'
+
+  local component
+
+  yellow_underlined='\e[4;33m'
```

Comment:
I'm pretty sure we have similar existing constants in one of the other library files - probably don't need to redefine them here.